source/docs/software/advanced-controls/controllers/bang-bang.rst

```diff
@@ -0,0 +1,65 @@
+Bang-Bang Control with BangBangController
+=========================================
+
+.. warning:: Bang-bang control is an extremely aggressive algorithm that relies on response asymmetry to remain stable.  Be *absolutely certain* that your motor controllers have been set to "coast mode" before attempting to control them with a bang-bang controller, or else the braking action will fight the controller and cause potentially destructive oscillation.
+
+`"Bang-bang" control <https://en.wikipedia.org/wiki/Bang%E2%80%93bang_control>`__ is a control strategy that employs only two states: on and off.  This is roughly equivalent to a P loop with infinite gain.
+
+This may initially seem like a poor control strategy, as PID loops are known to become unstable as the gains become large.  However, unlike an ordinary P loop, a bang-bang controller can easily be made *asymmetric* - that is, the controller can be made to turn on when the process variable is below the setpoint, but to do nothing otherwise.  This allows the control effort in the forward direction to be made as large as possible without risking destructive oscillations as the control loop tries to correct a resulting overshoot.
+
+When controlling the velocity of high-inertia mechanisms under varying loads (like a shooter flywheel), a bang-bang controller can yield faster recovery time and thus better/more consistent performance than a proportional controller.
```

Comment:
I think there should be a more explicit recommendation for high inertia and recommendation against low inertia mechanisms